mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/OrderEventsDaoImpl.java

```diff
@@ -44,19 +54,62 @@ public Future<RowSet<Row>> save(OrderAuditEvent orderAuditEvent, String tenantId
     return promise.future();
   }
 
+  @Override
+  public Future<OrderAuditEventCollection> getAcquisitionOrderAuditEventById(String orderId, int limit, int offset, String tenantId) {
+    Promise<RowSet<Row>> promise = Promise.promise();
+    try {
+      String jobTable = formatDBTableName(tenantId, TABLE_NAME);
+      String query = format(GET_BY_ID_SQL, jobTable);
+      Tuple queryParams = Tuple.of(UUID.fromString(orderId), limit, offset);
+      pgClientFactory.createInstance(tenantId).selectRead(query, queryParams, promise);
+    } catch (Exception e) {
+      LOGGER.error("Error getting OrderAuditEvent by id", e);
+      promise.fail(e);
+    }
+    return promise.future().map(rowSet -> rowSet.rowCount() == 0 ? new OrderAuditEventCollection().withTotalItems(0)
+      : mapRowToListOfOrderEvent(rowSet));
+
+  }
+
   private void makeSaveCall(Promise<RowSet<Row>> promise, String query, OrderAuditEvent orderAuditEvent, String tenantId) {
     try {
       pgClientFactory.createInstance(tenantId).execute(query, Tuple.of(orderAuditEvent.getId(),
         orderAuditEvent.getAction(),
         orderAuditEvent.getOrderId(),
         orderAuditEvent.getUserId(),
         orderAuditEvent.getUserName(),
-        orderAuditEvent.getEventDate(),
-        orderAuditEvent.getActionDate(),
-        new JsonObject(orderAuditEvent.getOrderSnapshot())), promise);
+        LocalDateTime.ofInstant(orderAuditEvent.getEventDate().toInstant(),ZoneId.systemDefault()),
+        LocalDateTime.ofInstant(orderAuditEvent.getActionDate().toInstant(),ZoneId.systemDefault()),
+        orderAuditEvent.getOrderSnapshot()), promise);
     } catch (Exception e) {
       LOGGER.error("Failed to save record with Id {} in to table {}", orderAuditEvent.getId(), TABLE_NAME, e);
       promise.fail(e);
     }
   }
+
+  private OrderAuditEventCollection mapRowToListOfOrderEvent(RowSet<Row> rowSet) {
+    OrderAuditEventCollection orderAuditEventCollection = new OrderAuditEventCollection();
+    rowSet.iterator().forEachRemaining(row -> {
+      orderAuditEventCollection.getOrderAuditEvents().add(mapRowToOrderEvent(row));
+    });
+    orderAuditEventCollection.setTotalItems(orderAuditEventCollection.getOrderAuditEvents().size());
```

Comment:
TotalItems should represent how many records exists in DB, not how many returned for particular offset and limit.
It's needed to know how many requests need to build to retrieve all data for particular order from database

Reply:
fixed